### PR TITLE
 [Sparse] Polish softmax.py.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/sparse/softmax.py [hunk 1]
```diff
@@ -8,18 +8,13 @@
...
-    """Apply row-wise softmax to the non-zero entries of the sparse matrix.
+def softmax(input: SparseMatrix) -> SparseMatrix:
+    """Apply row-wise softmax to the non-zero elements of the sparse matrix.
 
-    If :attr:`A.val` takes shape :attr:`(nnz, D)`, then the output matrix
-    :attr:`A'` and :attr:`A'.val` take the same shape as :attr:`A` and
-    :attr:`A.val`. :attr:`A'.val[:, i]` is calculated based on
-    :attr:`A.val[:, i]`.
-
-    Parameters
```
[the remark this answers is not in the document]
Comment (jermainewang) [reply python/dgl/sparse/softmax.py:19]: PyTorch just adds a link to the corresponding functional API. https://pytorch.org/docs/stable/generated/torch.Tensor.sum.html?highlight=sum#torch.Tensor.sum

[the remark this answers is not in the document]
Comment (jermainewang) [reply python/dgl/sparse/softmax.py:19]: In this way, the parameter section should be retained.

File: python/dgl/sparse/softmax.py [hunk 2]
```diff
@@ -8,18 +8,13 @@
 __all__ = ["softmax"]
 
 
-def softmax(A: SparseMatrix) -> SparseMatrix:
-    """Apply row-wise softmax to the non-zero entries of the sparse matrix.
+def softmax(input: SparseMatrix) -> SparseMatrix:
```
Comment (mufeili) [inline python/dgl/sparse/softmax.py:11]: Are we making the change for all usage of `A` across APIs?

File: python/dgl/sparse/softmax.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (frozenbugs) [reply python/dgl/sparse/softmax.py:19]: Updated as discussed.

File: python/dgl/sparse/softmax.py (hunk 2 above)
Comment (frozenbugs) [reply python/dgl/sparse/softmax.py:11]: Yes, for APIs on a single SparseMatrix and link to the class method, we use input.

File: python/dgl/sparse/softmax.py [hunk 3]
```diff
@@ -35,8 +35,8 @@ def softmax(A: SparseMatrix) -> SparseMatrix:
     >>> col = torch.tensor([1, 2, 2, 0])
     >>> nnz = len(row)
     >>> val = torch.arange(nnz).float()
-    >>> A = from_coo(row, col, val)
-    >>> softmax(A)
+    >>> A = dglsp.from_coo(row, col, val)
+    >>> dglsp.softmax(A)
```
Comment (mufeili) [inline python/dgl/sparse/softmax.py:39]: missing `import dgl.sparse as dglsp`

**Review by mufeili — APPROVED** [review]
one minor comment

File: python/dgl/sparse/softmax.py (hunk 3 above)
Comment (frozenbugs) [reply python/dgl/sparse/softmax.py:39]: Since many other place in Sparse library are assuming the import exists, so I am not adding it in this PR. If we believe we need to import, we can add them in one go.
